ddtrace/appsec/_asm_request_context.py

```diff
@@ -319,12 +319,26 @@ def asm_request_context_manager(
     The ASM context manager
     """
     if config._appsec_enabled:
-        resources = _DataHandler()
-        asm_request_context_set(remote_ip, headers, headers_case_sensitive, block_request_callable)
+        resources = _on_context_started(remote_ip, headers, headers_case_sensitive, block_request_callable)
         try:
             yield resources
         finally:
-            resources.finalise()
-            core.set_item("asm_env", None)
+            _on_context_ended(resources)
     else:
         yield None
+
+
+def _on_context_started(remote_ip, headers, headers_case_sensitive, block_request_callable):
+    resources = _DataHandler()
```

Comment:
Ah I think I get it, since `_DataHandler` uses `ASM_Environment`, which uses `core` under the hood, it's all just basically using core to manage the context.

Reply:
You got it!